lib/get-pattern-files.js

```diff
@@ -0,0 +1,166 @@
+'use strict';
+
+var path = require('path'),
+  fs = require('fs'),
+  utility = require('./utility');
+
+/*
+ * Gets details on an html pattern (and its supporting files) coming from a Pattern Library-structured directory
+ *
+ * @param {Array} paths  path info on pattern
+ * @param {Object} options  pattern-importer options
+*/
+exports.getPatternFiles = function (paths, options) {
+
+  // open the individual pattern's data file
+  var patternYml = fs.readFileSync(path.join(paths.folder, options.dataFileName), {encoding:'utf8'});
+
+  // create an object to store data about this pattern's files
+  var patternFiles = {};
+  patternFiles.data = {};
+
+  // get metadata from pattern.yml file
+  var patternObject = utility.convertYamlToObject(patternYml);
+
+  // get the pattern's category directory
+  patternFiles.patternCategoryPath = utility.getCategoryPath(patternObject, options);
+
+  // determine the pattern's template destination directory
+  patternFiles.patternTemplatePath = path.resolve(path.join(options.htmlTemplateDest,patternFiles.patternCategoryPath));
+
+  // determin the pattern's styles destination directory
+  patternFiles.patternScriptsPath = path.join(options.scriptsDest,patternFiles.patternCategoryPath);
+
+  // use defined template type or default to html
+  var patternTemplate = utility.getPatternTemplateName(patternObject, options);
+
+  var fileType = path.extname(patternTemplate).replace('.','');
+
+  var fileSrc = path.join(paths.folder,patternTemplate);
+
+  patternFiles.files = [];
+
+  if(patternTemplate){
+
+    // convert twig includes IF we are switching category directory titles in our destination directory
+    if(options.convertCategoryTitles && options.convertCategoryTitlesData && (options.templateEngine === 'twig') && (patternObject.twig)){
+      
+      // convert the include statements in our twig files
+      var twigContent = utility.convertTwigIncludes(options,fs.readFileSync(path.join(paths.folder,patternTemplate), {encoding:'utf8'}))
+
+      // our html template file destination
+      var fileDest = path.join(patternFiles.patternTemplatePath,paths.directory+'.twig');
+
+      // create the vinyl version
+      var file = utility.createNewFile(fileDest,path.resolve('./'),'',new Buffer(twigContent));
+      
+      // push to our files array
+      patternFiles.files.push(file);
+      
+    } else {
+      
+      // no conversion
+
+      // template file source
+      var fileSrc = path.join(paths.folder,patternTemplate);
+
+      // template file destination
+      var fileDest = path.join(patternFiles.patternTemplatePath,patternTemplate);
+
+      // create the vinyl version
+      var file = utility.createFile(fileSrc,path.resolve('./'),'','',fileDest);
```

Comment:
@sergesemashko this system now returns vinyl files to gulp and gulp does all the file writing work